pkg/loader/load_test.go

```diff
@@ -473,3 +475,38 @@ func (s *runSuite) TestShouldFlushWhenInputIsEmpty(c *check.C) {
 	time.Sleep(10 * time.Millisecond)
 	c.Assert(executed, check.HasLen, 7)
 }
+
+type markSuccessesSuite struct{}
+
+var _ = check.Suite(&markSuccessesSuite{})
+
+func (ms *markSuccessesSuite) TestShouldSetFinishTS(c *check.C) {
+	origF := fGetFinishTS
+	defer func() {
+		fGetFinishTS = origF
+	}()
+	fGetFinishTS = func(*sql.DB) int64 {
+		return 88881234
+	}
+	loader := &loaderImpl{downstreamTp: tidbTp, successTxn: make(chan *Txn, 64)}
+	loader.markSuccess([]*Txn{}...) // Make sure it won't break when no txns are passed
+	txns := []*Txn{
+		{Metadata: 1},
+		{Metadata: 3},
+		{Metadata: 5},
+	}
+	loader.markSuccess(txns...)
+	c.Assert(txns[len(txns)-1].FinishTS, check.Equals, fGetFinishTS(nil))
+
+	originUpdateLastFinishTSInterval := updateLastFinishTSInterval
+	updateLastFinishTSInterval = 0
+	defer func() {
+		updateLastFinishTSInterval = originUpdateLastFinishTSInterval
+	}()
+	txns = []*Txn{
+		{Metadata: 7},
+		{Metadata: 8},
+	}
+	loader.markSuccess(txns...)
+	c.Assert(txns[len(txns)-1].FinishTS, check.Equals, int64(88881234))
```

Comment:
The original assertion is to make sure that when we call `markSuccesses` again no `FinishTS` are set, so it should be `0`.

Reply:
updated in https://github.com/pingcap/tidb-binlog/pull/670/commits/8b1e5bde976da233ddbcdd4ef36291ef59b421ba